pyk/src/pyk/cterm/show.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from ..kast.inner import KApply, KSort, KToken, flatten_label, top_down
+from ..kast.manip import free_vars, minimize_term
+from ..kast.prelude.k import DOTS
+from ..kast.pretty import PrettyPrinter
+from .cterm import CTerm
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Iterable
+    from typing import Final
+
+    from ..kast.inner import KInner
+    from ..kast.outer import KDefinition, KFlatModule
+    from ..kast.pretty import SymbolTable
+
+_LOGGER: Final = logging.getLogger(__name__)
+
+
+class CTermShow(PrettyPrinter):
```

Comment:
IIUC `CTermShow` is for printing `CTerm` instances, whereas `PrettyPrinter` is for printing `KAst` instances, so in this case composition is sufficient I think:

```suggestion
@final
class CTermShow:
    printer: PrettyPrinter
```

Then you can probably remove

```
definition
extra_unparsing_modules
patch_symbol_table
unalias
sort_collections
```

from the `__init__` argument list.

Reply:
In fact, I'd reduce coupling between these classes a bit further: all this class requires is a `Callable[[KInner], str]]`. So I'd store that as an attribute, and provide something like

```python3
@staticmethod
def from_pretty_printer(printer: PrettyPrinter) -> CTermShow: ...
```

that creates the appropriate closure from the `PrettyPrinter`. Similarly, we can provide

```python3
@staticmethod
def from_formatter(formatter: Formatter) -> CTermShow: ...
```

and other implementations as needed.

Reply:
Ok, I've done this. Instead of providing static methods on `CTermShow` instead I've added a `property PrettyPrinter.printer` and `property Formatter.printer`, so it's easy to get the needed `Callable[[KInner], list[str]]` directly out of those classes for constructing the `CTermShow`.

Reply:
> Instead of providing static methods on `CTermShow` instead I've added a property `PrettyPrinter.printer` and property `Formatter.printer`

* I think the interface should be `Callable[[KInner], str]` as it is simpler (`Formatter` is already a `Callable[[KInner], str]` so it could even be passed directly). Also, since the `split` has to be done in both providers, it could be done internally `CTermShow` where necessary (by calling a helper `_print_lines` or similar).
* I think the `staticmethod` approach is cleaner: when reading the code it is immediately obvious what the intention is. On the other hand, if you add `printer` methods, in a way, you introduce a dependency in the wrong direction.

Reply:
I switched to `Callable[[KInner], str]`, which makes it so that I can just pass `PrettyPrinter.print` and `Formatter.print` directly as teh arguments to `CTermShow` constructor. Because this is pretty straightforward, I didn't add the `@staticmethod def from_...` options (it's not really any shorter).

Reply:
One benefit of `list[str]` as a return is that we can represent the empty return (no lines returned), whereas `''` always it at least one line returned.

Reply:
> Because this is pretty straightforward, I didn't add the `@staticmethod`

Makes sense.

> I can just pass `PrettyPrinter.print` and `Formatter.print`

For `Formatter`, you can even pass the instance directly. The same can be achieved for `PrettyPrinter` by implementing `__call__` with the right signature.